src/lib/Badge/index.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { FunctionComponent } from 'react'
+import clsx from 'clsx'
+
+import styles from './badge.module.scss'
+
+interface Props {
+  children: string
+  type?: 'primary' | 'secondary'
+  href?: string
+  className?: string
+  [rest: string]: unknown // ...rest property
+}
+
+const Badge: FunctionComponent<Props> = ({
+  children,
+  width = '',
+  height = '',
+  type = 'primary',
+  href = '',
+  className = '',
+  ...rest
+}) => {
+  const badgeClassName = clsx(styles.badge, styles[type], href && styles.link, className)
+  const badgeSizeStyles = { width: `${width}`, height: `${height}` }
+
+  if (href) {
```

Comment:
What if you want the badge to be a clickable button? I have an idea that can get rid of this if statement: Make a prop called `component`. Then you can pass in 'span', 'div', 'a', 'button' or whatever. Since you are passing the ...rest prop, you can add any attribute you need. Material UI does it here https://github.com/mui-org/material-ui/blob/next/packages/material-ui/src/Badge/Badge.js#L176

Reply:
I'm trying to figure out how to do this now. I still need to type it and I'm not sure what to type it as.